app/src/main/java/org/simple/clinic/util/AndroidExtensions.kt

```diff
@@ -64,3 +64,14 @@ private fun Configuration.isLocaleAlreadyOverriden(): Boolean {
     else -> false
   }
 }
+
+inline fun Dialog.onBackPressed(crossinline backPressed: () -> Unit) {
+  setOnKeyListener { _, keyCode, event ->
+    if (event.action == KeyEvent.ACTION_UP && keyCode == KeyEvent.KEYCODE_BACK) {
```

Comment:
I feel like this condition `event.action == KeyEvent.ACTION_UP && keyCode == KeyEvent.KEYCODE_BACK` is being used in multiple places. Should we move this to an extension property, say, `KeyEvent.isBackPressEvent` or something?

Reply:
Actually, this is the only place we are using it now, I have removed the previous `onOverrideKey` extension. So for sheets that want to handle the back clicks manually, they need to use this extension.